File: Programmers/Lv.1/최소직사각형/86491.js
function solution(sizes) {
    let answer = 0;
    
    let maxW = Math.max(...sizes[0]);
    let maxH = Math.min(...sizes[0]);
    for (const size of sizes) {
        const [w, h] = [Math.max(...size), Math.min(...size)];
        if (w > maxW) maxW = w;
        if (h > maxH) maxH = h;
    }
    
    answer = maxW * maxH;
    
    return answer;
}

File: Programmers/Lv.1/최소직사각형/86491.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int maxW = Math.max(sizes[0][0], sizes[0][1]);
        int maxH = Math.min(sizes[0][0], sizes[0][1]);
        for (int[] size : sizes) {
            int w = Math.max(size[0], size[1]);
            int h = Math.min(size[0], size[1]);
            
            if (w > maxW) maxW = w;
            if (h > maxH) maxH = h;
        }
        
        answer = maxW * maxH;
        
        return answer;
    }
}
